phoenix-ios/phoenix-ios/views/receive/ReceiveLightningView.swift

```diff
@@ -888,10 +822,93 @@ struct ReceiveLightningView: View {
 		deepLinkManager.broadcast(DeepLink.liquiditySettings)
 	}
 	
+	func showInboundFeeWarning(_ warning: InboundFeeWarning) {
+		
+		smartModalState.display(dismissable: true) {
+			InboundFeeSheet(warning: warning)
+		}
+	}
+	
 	// --------------------------------------------------
 	// MARK: Utilities
 	// --------------------------------------------------
 	
+	func refreshInboundFeeWarning() {
+		log.trace("refreshInboundFeeWarning()")
+		
+		inboundFeeWarning = calculateInboundFeeWarning()
+	}
+	
+	func calculateInboundFeeWarning() -> InboundFeeWarning? {
+		
+		let availableForReceiveMsat = channels.map { $0.availableForReceive?.msat ?? Int64(0) }.sum()
+		let hasNoLiquidity = availableForReceiveMsat == 0
+		
+		let canRequestLiquidity = channels.contains { $0.isUsable }
```

Comment:
Same as above, there's a `canRequestLiquidity` helper method in `LocalChannelInfo` which does the same thing. Hopefully it is usable from Swift.

Reply:
Done in [7a36358](https://github.com/ACINQ/phoenix/pull/544/commits/7a36358f5310a08df2c68c6072154e4fe0e2d3fb)